scripts/gulp/src/plugins/util/getShorthandInfo.ts

```diff
@@ -74,26 +75,24 @@ const getShorthandInfo = (componentFile: t.File, componentName: string): Shortha
         });
 
         if (mappedProperty) {
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore
-          t.assertStringLiteral(mappedProperty.node.value);
-          mappedShorthandProp = (mappedProperty.node.value as t.StringLiteral).value;
+          assertStringLiteral(mappedProperty.node.value);
+          mappedShorthandProp = mappedProperty.node.value.value;
         }
       } else if (isShorthandExpression(componentName, path)) {
         implementsCreateShorthand = true;
 
-        const config = path.get('right.arguments.0') as NodePath<t.ObjectExpression>;
+        // why is explicit annotation + assertion needed ? https://github.com/microsoft/TypeScript/issues/47945
+        const config: NodePath = path.get('right.arguments.0') as NodePath;
+
         config.assertObjectExpression();
 
-        const mappedProperty = (config.node.properties as any[]).find((property: t.ObjectProperty) => {
-          return t.isIdentifier(property.key, { name: 'mappedProp' });
-        }) as t.ObjectProperty | undefined;
+        const mappedProperty = (config.node.properties as t.ObjectProperty[]).find(property => {
+          return isIdentifier(property.key, { name: 'mappedProp' });
+        });
 
         if (mappedProperty) {
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore
-          t.assertStringLiteral(mappedProperty.value);
-          mappedShorthandProp = (mappedProperty.value as t.StringLiteral).value;
+          assertStringLiteral(mappedProperty.value);
```

Comment:
now babel types are shipping proper types thus narrowing works like charm 🙌 - no need for manual asserts anymore